### DISCUSSION TITLE
Adding/Improving README.md file to folders.

### DISCUSSION THREAD

**Discussion by Lukong123** [networkx/networkx#5437] [Q&A] [answered]
Hello everyone, 
Some folders either lack a readme file or their readme file could be improved. This might help newcomers to know their way around and also the readme-files can be used in the documentation generation pipeline. Please what are your thoughts on this?

**Comment by rossbar** [comment, marked as answer]
Thanks for the suggestion @Lukong123 !

Typically, information about the *Python source files* is found in files themselves. For example, most modules (i.e. `.py` files) *should* have a module docstring at the very top of the file that generally describes how the module is organized and what can be found there. As an example, take a look at [the chordal module](https://github.com/networkx/networkx/blob/1e48dad5c6038b549d7a4a01f190ffd3765eb913/networkx/algorithms/chordal.py#L1-L7). It's not *a ton* of information, but it should provide a snapshot of what's in the module and how it's organized.

> can be used in the documentation generation pipeline.

Fortunately this is already the case - if you look at the online documentation for `chordal` it [should look familiar...](https://networkx.org/documentation/latest/reference/algorithms/chordal.html#chordal). Module docstrings are (or *should be*) automatically used in generating the html documentation. You're right that this is indeed a very convenient and important feature, as it helps guarantee that the documentation of the code stay in-sync.

All that said, I'm sure there are many opportunities to improve the situation. For example, I'm sure some modules are missing docstrings. There's also the issue of *packages* (i.e. folders that contain other modules and packages, like `algorithms/` for example) which typically documented with a different pattern.

Hopefully this information is useful in navigating the source code and documentation - please don't hesitate to followup with additional comments/thoughts!

**Reply by Lukong123** [reply]
Okay, thanks, sir.
